Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,34 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python?(py)
+        requirements.any? do |req|
+          req.name == py && (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == py && (dep.required? || build.with?(dep))
+          end
```

Comment:
I have to say that I dislike the duplication in these `<whatever>.any? do` blocks and even more so that those multi-line blocks are joined via a Boolean operation. Can you try and make this a bit nicer?